Disable aggregates (#40)
graphile/pg-aggregates: README.md

See [src/AggregateSpecsPlugin.ts](src/AggregateSpecsPlugin.ts) for examples and
more information.

## Disable aggregates

> The aggregates are created for all tables which really massively increases the type def in size.
> You can disable aggregates by default and enable them only for the tables you need.

Reviewer: ```suggestion
By default, aggregates are created for all tables. This significantly increases the size of your
GraphQL schema, and could also be a security (DoS) concern as aggregates can be expensive. We
recommend that you use the `disableAggregatesByDefault: true` option to disable aggregates by
default, and then enable them only for the tables you need:
```

```ts

> // Disable aggregates by default
> graphileBuildOptions: {
>   disableAggregatesByDefault: true,
> }

Reviewer: People don't necessarily know where to put `graphileBuildOptions`, so here's a slightly more complete example:

```suggestion
const middleware = postgraphile(DATABASE_URL, SCHEMAS, {
  // ...
  appendPlugins: [
    // ...
    PgAggregatesPlugin,
  ],
  
  graphileBuildOptions: {
    // Disable aggregates by default; opt each table in via the `@aggregates` smart tag
    disableAggregatesByDefault: true,
  },
});
```

```

Enable aggregates for a specific table:

> ```ts
> "my_schema.my_table": {
>   "tags": {
>     "aggregates": "on"
>   }
> }
> ```

Reviewer: ````suggestion
```json
"class": {
  "my_schema.my_table": {
    "tags": {
      "aggregates": "on"
    }
  }
}
```
````

or:

```sql
COMMENT ON TABLE my_schema.my_table IS E'@aggregates on';
```

You also can keep aggregates enabled by default, but disable aggregates for specific tables:

> ```ts
> "my_schema.my_table": {
>   "tags": {
>     "aggregates": "off"
>   }
> }
> ```

Reviewer: ````suggestion
```json
"class": {
  "my_schema.my_table": {
    "tags": {
      "aggregates": "off"
    }
  }
}
```
````

See [src/AggregateSpecsPlugin.ts](src/AggregateSpecsPlugin.ts) for examples and
more information.

## Disable aggregates

By default, aggregates are created for all tables. This significantly increases the size of your
GraphQL schema, and could also be a security (DoS) concern as aggregates can be expensive. We
recommend that you use the `disableAggregatesByDefault: true` option to disable aggregates by
default, and then enable them only for the tables you need:

```ts
const middleware = postgraphile(DATABASE_URL, SCHEMAS, {
  // ...
  appendPlugins: [
    // ...
    PgAggregatesPlugin,
  ],
  
  graphileBuildOptions: {
    // Disable aggregates by default; opt each table in via the `@aggregates` smart tag
    disableAggregatesByDefault: true,
  },
});
```

Enable aggregates for a specific table:

```json
"class": {
  "my_schema.my_table": {
    "tags": {

>       "aggregates": "off"

Reviewer: ```suggestion
      "aggregates": "on"
```

    }
  }
}
```

or:

```sql
COMMENT ON TABLE my_schema.my_table IS E'@aggregates on';
```

You also can keep aggregates enabled by default, but disable aggregates for specific tables:

```json
"class": {
  "my_schema.my_table": {
    "tags": {

>       "aggregates": "on"

Reviewer: ```suggestion
      "aggregates": "off"
```
